### PR TITLE
Generalize the EvalFile net option detection

### PR DESCRIPTION
**Pull request by ppigazzini**
Stockfish will use more than one net with
https://github.com/official-stockfish/Stockfish/pull/4915

Get all the couples "option" and "net names" from the output of the
`stockfish uci` command.

Raise worker version to 227 (also server side)

### CODE REVIEW

File: worker/games.py [hunk 1]
```diff
@@ -1341,17 +1341,14 @@ def parse_options(s):
...
             )
 
-    # Add EvalFile with full path to cutechess options, and download the networks if missimg.
-    net_base = required_net(base_engine)
-    if net_base:
-        base_options = base_options + ["option.EvalFile={}".format(net_base)]
-    net_new = required_net(new_engine)
-    if net_new:
-        new_options = new_options + ["option.EvalFile={}".format(net_new)]
+    # Add EvalFile* with full path to cutechess options, and download the networks if missimg.
```
Comment (ddobbelaere) [inline worker/games.py:1344]: s/missimg/missing/ ;)

**Comment by mstembera** [comment]
@ppigazzini Thank you for this.  Is this the last thing needed on the fishtest side before https://github.com/official-stockfish/Stockfish/pull/4915 can be merged?

**Comment by mstembera** [comment]
@ppigazzini Can you please commit this so we can proceed with https://github.com/official-stockfish/Stockfish/pull/4915 ?  Ty!

**Comment by peregrineshahin** [comment]
Maybe ppigazzini realized a big flaw in this PR and still thinking about the solution ;) ..

**Comment by Disservin** [comment]
btw we should also update the memory requirements even though it's not much I guess.
The second net is 3MB, the current main net is estimated to be around 80MB in fishtest code which I think is still an old value, because the net file is around ~64MB. 

Also fishtest and the worker are out of sync regarding that. 
https://github.com/official-stockfish/fishtest/blob/master/server/fishtest/rundb.py#L1108
https://github.com/official-stockfish/fishtest/blob/master/worker/worker.py#L800

the server code is missing the 16MB ?

**Comment by ppigazzini** [comment]
Tested with workers: msys2, linux (python 3.6.8) with:
 - sf_11 vs master
 - sf_classical vs master
 - sf_12 vs master
 - SF_NNUE vs master
 - DualNNUE-1100-pr vs master
 
DualNNUE-1100-pr requires `'option.EvalFile=nn-0000000000a0.nnue', 'option.EvalFileSmall=nn-c01dc0ffeede.nnue'`

image

**Comment by ppigazzini** [comment]
Triggered the worker update, of course.

**Comment by Disservin** [comment]
Thanks. I made a PR for the according Stockfish changes.

**Pull request merged**